packages/salesforcedx-utils-vscode/src/commands/commandletExecutors.ts

```diff
@@ -219,9 +219,11 @@ export abstract class LibraryCommandletExecutor<T>
         measurements
       );
     } catch (e) {
-      telemetryService.sendException(e.name, e.message);
-      notificationService.showFailedExecution(this.executionName);
-      channelService.appendLine(e.message);
+      if (e instanceof Error) {
```

Comment:
This is something to watch out for.  I did a search, and it looks like Error object are being thrown everywhere, except 3 places:
packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts, line 40
packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts, line 68
packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts, line 65

If these 3 places were updated to `throw new Error(string)` instead of `throw string`, the check for `e instanceof Error` wouldn't bee needed.

Reply:
Good call. Yeah I think at least one of those places I originally checked for an Error and ran into the issue of it being a string.  I'll add an item to the list for having the linter only allow us to throw errors. I started adding this check b/c with the typescript update they tightened up on assuming Errors in catch blocks (see [here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#defaulting-to-the-unknown-type-in-catch-variables---useunknownincatchvariables).  I ended up deciding there were too many touches for this and ran into the issue in faux generate so I turned off the check in the tsconfig and added a note to follow up.  Will also ensure we have a ticket to address this.